### PR TITLE
Improve Gathering and Updating Art

### PR DESCRIPTION
**Pull request by wagnered**
1. Added capability to import pictures from FLAC fies.

2. Modified `Art::get_from_source($data, $type)` to get song image from `metaTagCollectorModule` class.

3. Improved ability to update(replace) existing song images.

4. Added capability of getting mime type from actual image signature.

### CODE REVIEW

**Comment by lachlan-00** [comment]
few issues on this one, i think it's mostly just typo
https://scrutinizer-ci.com/g/ampache/ampache/inspections/9b3e9f6d-1a5e-40dc-b715-9353fd77bf7d/issues/

**Comment by wagnered** [comment]
Thanks,

Sometimes when I'm tired, I get mentally dyslexic.

The scrutinizer is referring to stuff that no longer exists in SongId3TagWriter.php:  I reverted the class to the previous version.

Seems to me that scrutinizer messages should be added to the PR comments.

Properly writing song data to both mp3 and flac tags is my next project.

File: src/Repository/Model/Album.php [hunk 1]
```diff
@@ -944,7 +944,7 @@ public function update(array $data)
             }
 
             if (empty($data['album_artist_name']) && !empty($album_artist) && $album_artist != $this->album_artist) {
-                self::update_field('album_artist', $album_artist, $this->id);
```
Comment (lachlan-00) [inline src/Repository/Model/Album.php:947]: just need to revert this one back to album_artist

File: src/Repository/Model/Art.php [hunk 2]
```diff
@@ -379,7 +379,7 @@ public function insert($source, $mime = '')
             foreach ($songs as $song_id) {
                 $song   = new Song($song_id);
                 $song->format();
-                $description = ($this->type == 'artist') ? $song->f_artist_full : $object->f_name;
```
Comment (lachlan-00) [inline src/Repository/Model/Art.php:382]: back to f_name too

**Comment by lachlan-00** [comment]
cool, time to test

**Comment by lachlan-00** [comment]
perfect. seems to be even faster too

**Comment by lachlan-00** [comment]
just found out why it's so fast
```
2021-07-01T16:55:24+10:00 [ampache] (log.lib) -> [Runtime Error] Illegal string offset 'raw' in file /var/www/music/src/Module/Application/Art/FindArtAction.php(121)
2021-07-01T16:55:24+10:00 [ampache] (log.lib) -> [Runtime Error] Illegal string offset 'mime' in file /var/www/music/src/Module/Application/Art/FindArtAction.php(121)

```

**Comment by lachlan-00** [comment]
i'm not going to be able to fix this until next week but just so i can remember this is my query for finding missing id's.

Luckily song never really gets deleted so we can repair based on that

```
# missing artist & album (I'm missing 26)
SELECT * FROM object_count WHERE object_type = 'song' AND count_type = 'stream' AND `date` NOT IN (select date from object_count WHERE count_type = 'stream' and (object_type = 'album' OR object_type = 'artist'))
ORDER BY date DESC;

# missing artist (I'm missing 87)
SELECT * FROM object_count WHERE object_type = 'song'  AND count_type = 'stream' AND `date` NOT IN (select date from object_count WHERE count_type = 'stream' and object_type = 'artist')
ORDER BY date DESC;

# missing album (I'm missing 18307!!!!!!!)
SELECT * FROM object_count WHERE  count_type = 'stream' AND `date` NOT IN (select date from object_count WHERE count_type = 'stream' and object_type = 'album')
ORDER BY date DESC;
```

**Comment by lachlan-00** [comment]
* deleted comment from the wrong issue

**Comment by wagnered** [comment]
```
2021-07-01T16:55:24+10:00 [ampache] (log.lib) -> [Runtime Error] Illegal string offset 'raw' in file /var/www/music/src/Module/Application/Art/FindArtAction.php(121)
2021-07-01T16:55:24+10:00 [ampache] (log.lib) -> [Runtime Error] Illegal string offset 'mime' in file /var/www/music/src/Module/Application/Art/FindArtAction.php(121)
```

Ach!  When I closed the last PR and reverted, I lost all references to function `get_mime_from_image($data)` and the reconfigured return values.

Looking at it now.

**Comment by wagnered** [comment]
I just noticed on the Artist view that  the label tab has a blank image, although the Artist image on the upper right side is correct.

What is that image supposed to represent and how does it get linked to the artist and/or label id?

Thanks..

**Comment by wagnered** [comment]
Ok, I found the connections: Via `label_assoc` table.  I have yet to discover how an image gets imported into the db and linked to the label.    maybe my PR branch is screwed up.  I will checkout develop and give it a try.

**Comment by wagnered** [comment]
Looks like it's not my problem, so I'll clean up the changes I have and push them through.

**Comment by lachlan-00** [comment]
took that part out in 51bfe9001 for now

**Pull request merged**